### DISCUSSION TITLE
Change 'helmfile version' (and any other similar command) so that it doesn't reach out to GitHub or the Internet

### DISCUSSION THREAD

**Discussion by jcmcken** [helmfile/helmfile#1490] [Ideas]
For teams that have their own private networks and Helm registries, it's a nuisance if helmfile commands are trying to contact the Internet. For instance, ``helmfile version``, in our environment, prints a version string and then hangs until some TCP timeout. When I ``strace`` the process, I can see it trying to contact GitHub. This seems like an anti-pattern for something like a version command, which I think users expect to be a very simple, local command.

I'm not sure if there are any other commands that exhibit this same sort of issue. One common example (again, unsure if this applies to ``helmfile``) is the use of remote telemetry or debugging. For example, if the tool crashes, it could send a debug report or metrics. Typically tools that have such telemetry or debugging have some method of opting out, since these features cause significant problems in on-premise, airgapped environments.

So I guess my proposal depends on the exact nature of these remote calls -- is this something only present in ``version`` or is it spread into other aspects of the tool? 

* If the former, I propose either changing the default of ``helmfile version`` to not contact GitHub, or else provide a flag or environment variable or other global setting to disable it.
* If the latter, tools commonly implement a global opt-out environment variable or configuration so that you don't need to continuously opt-out every time you run the tool.

**Comment by yxxhero** [comment]
HELMFILE_UPGRADE_NOTICE_DISABLED
